### PR TITLE
Add env var to tune header writes delay

### PR DESCRIPTION
**Pull request by divan**
This PR serves as temporary solution to verify CPU/temperature impact from slowing down header writes to leveldb and collect data on real device. Once data is collected, this PR should be removed and proper/correct implementation should be made. See #432 for ideas.

It introduces new environment variable for this feature - `FEATURE_SYNC_DELAY` which sets the delay between writes in **microseconds**. Default is 0, meaning no delay.

### CODE REVIEW

**Review by tiabc — APPROVED** [review]
lgtm

File: vendor/github.com/ethereum/go-ethereum/core/headerchain.go [hunk 1]
```diff
@@ -40,6 +42,21 @@ const (
...
+var writeDelay time.Duration = 0
+
+func init() {
+	str, ok := os.LookupEnv("FEATURE_SYNC_DELAY")
+	if !ok {
+		return
+	}
+	delay, err := strconv.ParseInt(str, 10, 0)
+	if err != nil {
+		panic(fmt.Errorf("FEATURE_SYNC_DELAY value is invalid:", err))
```
Comment (adambabik) [inline vendor/github.com/ethereum/go-ethereum/core/headerchain.go:54]: `panic()` does not take a string as its argument?

File: vendor/github.com/ethereum/go-ethereum/core/headerchain.go [hunk 2]
```diff
@@ -271,6 +292,11 @@ func (hc *HeaderChain) InsertHeaderChain(chain []*types.Header, writeHeader WhCa
 			stats.ignored++
 			continue
 		}
+
+		// introduce delay for the expensive write to
+		// trade off sync time with CPU usage
+		time.Sleep(writeDelay)
+
 		if err := writeHeader(header); err != nil {
```
Comment (adambabik) [inline vendor/github.com/ethereum/go-ethereum/core/headerchain.go:300]: Is this method sync or async? Does it try to do IO operation directly?

**Review by adambabik — APPROVED** [review]

File: vendor/github.com/ethereum/go-ethereum/core/headerchain.go (hunk 1 above)
Comment (divan) [reply vendor/github.com/ethereum/go-ethereum/core/headerchain.go]: @adambabik hence `fmt.Errorf()` call :)

Comment (divan) [reply vendor/github.com/ethereum/go-ethereum/core/headerchain.go:54]: @adambabik I think it takes interface{}, but error works just fine.

File: vendor/github.com/ethereum/go-ethereum/core/headerchain.go (hunk 2 above)
Comment (divan) [reply vendor/github.com/ethereum/go-ethereum/core/headerchain.go:300]: I think it's synchronous, that's why introducing delay has an effect.

**Comment by oskarth** [comment]
@divan @adambabik  Not sure if you see stuff in status-react, so I'm double posting here. Seems like there is an issue with the artifact uploaded - can you check? https://github.com/status-im/status-react/pull/2509#issuecomment-346556732

**Pull request merged**